util/fibers/epoll_socket.cc

```diff
@@ -166,22 +166,18 @@ bool EpollSocket::AsyncReq::Run(int fd, bool is_send) {
   res = is_send ? sendmsg(fd, &msg, MSG_NOSIGNAL) : recvmsg(fd, &msg, 0);
 
   if (res > 0) {
-    cb(res);
-    return true;
+    return {true, res};
```

Comment:
what is the reason for moving cb() outside of AsyncReq::Run?

Reply:
This is the fix. If `cb()` calls again `AsyncWrite` it will `CHECK FAIL` because the `AsyncReq` is not `nullptr`. We have to reset the `async request` before we call the callback and to do that, we need to move it outside of the object we are about to deallocate